src/CMakeLists.txt

```diff
@@ -0,0 +1,93 @@
+set(ASSEMBLYLINE_SRC_DIR ${ASSEMBLYLINE_ROOT_DIR}/src CACHE INTERNAL "ASSEMBLYLINE_SRC_DIR")
+set(ASSEMBLYLINE_MAN_DIR ${ASSEMBLYLINE_ROOT_DIR}/man CACHE INTERNAL "ASSEMBLYLINE_MAN_DIR")
+
+include (GenerateExportHeader)
+
+set(
+    HEADER_FILES
+
+    ${ASSEMBLYLINE_SRC_DIR}/assembler.h
+    ${ASSEMBLYLINE_SRC_DIR}/common.h
+    ${ASSEMBLYLINE_SRC_DIR}/encoder.h
+    ${ASSEMBLYLINE_SRC_DIR}/enums.h
+    ${ASSEMBLYLINE_SRC_DIR}/instr_parser.h
+    ${ASSEMBLYLINE_SRC_DIR}/instruction_data.h
+    ${ASSEMBLYLINE_SRC_DIR}/instructions.h
+    ${ASSEMBLYLINE_SRC_DIR}/parser.h
+    ${ASSEMBLYLINE_SRC_DIR}/prefix.h
+    ${ASSEMBLYLINE_SRC_DIR}/reg_parser.h
+    ${ASSEMBLYLINE_SRC_DIR}/registers.h
+    ${ASSEMBLYLINE_SRC_DIR}/tokenizer.h
+)
+
+set(
+    SOURCE_FILES
+
+    ${ASSEMBLYLINE_SRC_DIR}/assembler.c
+    ${ASSEMBLYLINE_SRC_DIR}/assemblyline.c
+    ${ASSEMBLYLINE_SRC_DIR}/encoder.c
+    ${ASSEMBLYLINE_SRC_DIR}/instr_parser.c
+    ${ASSEMBLYLINE_SRC_DIR}/instructions.c
+    ${ASSEMBLYLINE_SRC_DIR}/parser.c
+    ${ASSEMBLYLINE_SRC_DIR}/prefix.c
+    ${ASSEMBLYLINE_SRC_DIR}/reg_parser.c
+    ${ASSEMBLYLINE_SRC_DIR}/registers.c
+    ${ASSEMBLYLINE_SRC_DIR}/tokenizer.c
+)
+
+configure_file(
+    "${ASSEMBLYLINE_SRC_DIR}/config.h.in"
+    "${ASSEMBLYLINE_SRC_DIR}/config.h"
+)
+
+add_library(AssemblyLine SHARED ${SOURCE_FILES})
+
+if (MSVC)
+    add_compile_options(/W3 /WX /Wall /sdl /guard:cf /guard:ehcont /CETCOMPAT)
+    set(
+        LINKING_LIBS
+
+        kernel32.lib
+        ntdll.lib
+    )
+else()
+    add_compile_definitions(_GNU_SOURCE=1)
+    add_compile_options(-Wall -Wextra -std=gnu99 -fPIC)
+
+    if(DEBUG)
+        add_compile_options(-ggdb)
+    endif(DEBUG)
+
+    set(
+        LINKING_LIBS
+
+        m
+    )
+endif()
+
+#
+# Linking directives
+#
+target_include_directories(AssemblyLine PUBLIC ${ASSEMBLYLINE_SRC_DIR}/)
+target_sources(AssemblyLine PRIVATE ${ASSEMBLYLINE_SRC_DIR}/)
+
+GENERATE_EXPORT_HEADER(
+    AssemblyLine
+    BASE_NAME AssemblyLine
+    EXPORT_MACRO_NAME AssemblyLine_EXPORT
+    EXPORT_FILE_NAME assemblyline_export.hpp
+    STATIC_DEFINE assemblyline_built_as_static
+)
+
+target_link_libraries(AssemblyLine PUBLIC ${LINKING_LIBS})
+
+#
+# Install directives
+#
+install(TARGETS AssemblyLine DESTINATION lib)
+install(FILES ${HEADER_FILES} DESTINATION include)
```

Comment:
```suggestion
install(FILES assemblyline.h DESTINATION include)
```

Reply:
Is that what you'd like me to confirm ? sure. Or would we want to change the variable `HEADER_FILES` instead (I don't know if/where it is used elsewhere)